### PR TITLE
Update tool versions for M. tuberculosis variant analysis tutorial

### PR DESCRIPTION
**Pull request by pvanheus**
Reviewing the _M. tuberculosis variant analysis tutorial_ I noticed that a lot of the tools mentioned are quite old versions. This PR updates the tool versions to the latest versions of each tool used in the tutorial and does similar (using `planemo autoupdate`) for the included workflow. It also changes the Kraken2 test: since the name of the Kraken2 standard database is not consistent, this test now just checks that Kraken2 ran successfully.

### CODE REVIEW

**Comment by wm75** [comment]
@pvanheus added an intermediate commit for reviewing purposes only. Will revert before merging.

**Comment by pvanheus** [comment]
> @pvanheus added an intermediate commit for reviewing purposes only. Will revert before merging.

Thanks. The `planemo autoupdate` should probably get an option for indent level.

**Comment by wm75** [comment]
@pvanheus quick question. For some reason *snippy* did not get updated in the workflow file. Is there a good reason for that or can I just do this manually while I'm on it?

**Comment by wm75** [comment]
`toolshed.g2.bx.psu.edu/repos/iuc/tbprofiler/tb_profiler_profile/4.1.1+galaxy0` is currently not available at .org and .org.au
You may want to convince @mvdbeek and @Slugger70 to update to the latest version of that tool. Both are at `3.0.8+galaxy0`

**Comment by wm75** [comment]
Same for `toolshed.g2.bx.psu.edu/repos/iuc/tbvcfreport/tbvcfreport/0.1.10+galaxy0`.
Both .org and .org.au only have `0.1.8+galaxy0`.

**Comment by wm75** [comment]
`- *"Produce Standalone Instance"*: `Yes``
This simple boolean has been reworked in the latest jbrowse version. Please adjust.

**Comment by wm75** [comment]
`toolshed.g2.bx.psu.edu/repos/iuc/jvarkit_wgscoverageplotter/jvarkit_wgscoverageplotter/20201223+galaxy0` - can't find the tool on usegalaxy.org.au. It's only used in he additional samples exploration part of the tutorial, but might still be good to have it @Slugger70

**Comment by wm75** [comment]
Ok, went through everything once. @pvanheus can you please double check that all numbers you are giving in *Solution* sections are still what people will see after all those tool updates?

**Comment by wm75** [comment]
... and one additional suggestion for (a later) improvement:
currently you're asking users to download:
```
https://zenodo.org/record/3960260/files/004-2_1.fastq.gz
https://zenodo.org/record/3960260/files/004-2_2.fastq.gz
https://zenodo.org/record/3960260/files/Mycobacterium_tuberculosis_ancestral_reference.gbk
https://zenodo.org/record/3960260/files/MTB_ancestor_reference.fasta
https://zenodo.org/record/3960260/files/Mycobacterium_tuberculosis_h37rv.ASM19595v2.45.chromosome.Chromosome.gff3
```
Instead of getting a genbank *and* a fasta file you could explain how to extract fasta from genbank (with seqret from the emboss suite) just before the step that requires it. Would teach people an additional thing you can do right in Galaxy.

**Comment by pvanheus** [comment]
> Same for `toolshed.g2.bx.psu.edu/repos/iuc/tbvcfreport/tbvcfreport/0.1.10+galaxy0`. Both .org and .org.au only have `0.1.8+galaxy0`.

This tool - like with `tb_profiler_profile` - was only updated in the last few days. Perhaps we should give the other admins a week before merging this PR then?

**Comment by pvanheus** [comment]
> ... and one additional suggestion for (a later) improvement: currently you're asking users to download:
> 
> ```
> https://zenodo.org/record/3960260/files/004-2_1.fastq.gz
> https://zenodo.org/record/3960260/files/004-2_2.fastq.gz
> https://zenodo.org/record/3960260/files/Mycobacterium_tuberculosis_ancestral_reference.gbk
> https://zenodo.org/record/3960260/files/MTB_ancestor_reference.fasta
> https://zenodo.org/record/3960260/files/Mycobacterium_tuberculosis_h37rv.ASM19595v2.45.chromosome.Chromosome.gff3
> ```
> 
> Instead of getting a genbank _and_ a fasta file you could explain how to extract fasta from genbank (with seqret from the emboss suite) just before the step that requires it. Would teach people an additional thing you can do right in Galaxy.

Good idea... implemented in latest version of the PR.

**Comment by pvanheus** [comment]
> `- *"Produce Standalone Instance"*: `Yes`` This simple boolean has been reworked in the latest jbrowse version. Please adjust.

Addressed, thanks for spotting it.

**Comment by pvanheus** [comment]
> Ok, went through everything once. @pvanheus can you please double check that all numbers you are giving in _Solution_ sections are still what people will see after all those tool updates?

I've checked the solutions and adjusted as necessary.

File: topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md [hunk 1]
```diff
@@ -308,13 +309,18 @@ We could go through all of the variants in the VCF files and read them out of a
 
 > ### {% icon hands_on %} Hands-on: Run JBrowse
 >
-> 1. {% tool [JBrowse](toolshed.g2.bx.psu.edu/repos/iuc/jbrowse/jbrowse/1.16.8+galaxy1) %} {% icon tool %} with the following parameters
+> 1. Use {% tool [seqret](tool_id=toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: seqret84/5.0.0) %} {% icon tool %} to convert the Genbank format reference (`Mycobacterium_tuberculosis_ancestral_reference.gbk`) to FASTA format. Use the following parameters:
```
Comment (wm75) [inline topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md:312]: @hexylena a tool ID with a space in it! Will that work or should the space be encoded or sth?

Comment (hexylena) [reply topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md:312]: Eesh. 

1. remove `tool_id=`, that's unnecessary
2. It *should* work, but safer to write a `+`.
3. this never should have been allowed :| 

[I've tested it just now](https://usegalaxy.eu/?tool_id=toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: seqret84/5.0.0) and it seems to auto-replace with a `+`

Comment (hexylena) [reply topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md:312]: lol, markdown can't handle that space. Unsurprising.

Comment (wm75) [reply topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md]: So the space gets translated into `+` as you've said, but the `+` version fails to locate the tool then.
`%20` seems to be ok for markdown, but reaches Galaxy as a space and triggers `+` replacement just the same.
I'd love to test `%2520` but I'm not sure how to test that.

Comment (hexylena) [reply topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md]: Ah I was looking at the wrong PR, https://github.com/galaxyproject/galaxy/pull/13347 yeah, %20 is the right answer.

I'll add linting for it on our end.

Comment (wm75) [reply topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md]: > Ah I was looking at the wrong PR, https://github.com/galaxyproject/galaxy/pull/13347 yeah, %20 is the right answer.

Thanks, should work from looking at the generated html, yes. Guess was my editor that got me confused even further :)

**Review by wm75 — APPROVED** [review]
@pvanheus seems good to go to me.
Thanks for keeping the material updated.

There seems to be a related training ongoing on .eu right now by updating the material and have half of the participants use the updated version.
So lets wait till the weekend with merging this and maybe .org and .org.au will have updated their tools until then, too :)

**Review by wm75 — APPROVED** [review]

**Pull request merged**